src/test/java/org/jfrog/hudson/pipeline/integration/ITestUtils.java

```diff
@@ -0,0 +1,242 @@
+     * @param repoKey           - Repository key
+     */
+    static void assertNoArtifactsInRepo(Artifactory artifactoryClient, String repoKey) {
+        List<RepoPath> repoPaths = artifactoryClient.searches().repositories(repoKey).artifactsByName("*in").doSearch();
+        assertTrue(repoPaths.isEmpty());
+    }
+
+    /**
+     * Assert artifacts exist in repository.
+     *
+     * @param artifactoryClient - Artifactory java client
+     * @param repoKey           - Repository key
+     * @param expectedArtifacts - Expected artifacts
+     */
+    static void assertArtifactsInRepo(Artifactory artifactoryClient, String repoKey, Set<String> expectedArtifacts) {
+        List<RepoPath> repoPaths = artifactoryClient.searches().repositories(repoKey).artifactsByName("*in").doSearch();
+        Set<String> actualArtifacts = repoPaths.stream().map(RepoPath::getItemPath).collect(Collectors.toSet());
+        assertEquals(expectedArtifacts, actualArtifacts);
+    }
+
+    /**
+     * Get module from the build-info object and assert its existence.
+     *
+     * @param buildInfo  - Build-info object
+     * @param moduleName - Module name
+     * @return module from the build-info
+     */
+    static Module getAndAssertModule(Build buildInfo, String moduleName) {
+        assertNotNull(buildInfo);
+        assertNotNull(buildInfo.getModules());
+        Module module = buildInfo.getModule(moduleName);
+        assertNotNull(module);
+        return module;
+    }
+
+    /**
+     * Assert that the artifacts and dependencies lists are not empty in the input module.
+     *
+     * @param buildInfo  - Build info object
+     * @param moduleName - Module name
+     */
+    static void assertModuleContainsArtifactsAndDependencies(Build buildInfo, String moduleName) {
+        Module module = getAndAssertModule(buildInfo, moduleName);
+        assertTrue(CollectionUtils.isNotEmpty(module.getArtifacts()));
+        assertTrue(CollectionUtils.isNotEmpty(module.getDependencies()));
+    }
+
+    /**
+     * Assert that the artifacts list is not empty in the input module.
+     *
+     * @param buildInfo  - Build info object
+     * @param moduleName - Module name
+     */
+    static void assertModuleContainsArtifacts(Build buildInfo, String moduleName) {
+        Module module = getAndAssertModule(buildInfo, moduleName);
+        assertTrue(CollectionUtils.isNotEmpty(module.getArtifacts()));
+    }
+
+    /**
+     * Delete build in Artifactory.
+     *
+     * @param artifactoryClient - Artifactory java client
+     * @param buildName         - Build name to delete
+     */
+    static void deleteBuild(Artifactory artifactoryClient, String buildName) throws IOException {
+        artifactoryClient.restCall(new ArtifactoryRequestImpl()
+                .method(ArtifactoryRequest.Method.DELETE)
+                .apiUrl("api/build/" + encodeBuildName(buildName))
+                .addQueryParam("deleteAll", "1"));
```

Comment:
Question - Wouldn't you like this method to also delete the build artifacts as well?
I so, you should add **.addQueryParam("artifacts", "1"))**
as described [here](https://www.jfrog.com/confluence/display/RTF/Artifactory+REST+API#ArtifactoryRESTAPI-DeleteBuilds).